tools/ts-generator/types/test.ts

```diff
@@ -107,4 +107,108 @@ describe("Buffer functions test", function () {
     asserts.deepStrictEqual(result, array);
     asserts.strictEqual(offset, 19);
   });
+
+  it("should write varint in buffer", function () {
+    const buffer = Buffer.alloc(8);
+    const size = BF.writeVarintBuffer(BigInt(56000), buffer);
+    const [number, offset] = BF.readVarint(buffer, 0);
+    asserts.deepStrictEqual(number, BigInt(56000));
+    asserts.strictEqual(offset, 3);
+    asserts.strictEqual(size, 3);
+  });
+
+  it("should extend records", function () {
+    const crc = BF.extendRecords([
+      {
+        attributes: 0,
+        key: Buffer.from([]),
+        keyLength: 0,
+        length: 10,
+        offsetDelta: 0,
+        timestampDelta: BigInt(0),
+        value: Buffer.from([54, 65, 73, 74]),
+        valueLen: 4,
+        headers: [],
+      },
+    ]);
+    const expectedCrc = 2892331115;
+    asserts.strictEqual(crc, expectedCrc);
+  });
+
+  it("should extend records with seed", function () {
+    const seed = 321123434;
+    const crc = BF.extendRecords(
+      [
+        {
+          attributes: 0,
+          key: Buffer.from([]),
+          keyLength: 0,
+          length: 10,
+          offsetDelta: 0,
+          timestampDelta: BigInt(0),
+          value: Buffer.from([54, 65, 73, 74]),
+          valueLen: 4,
+          headers: [],
+        },
+      ],
+      seed
+    );
+    const unexpectedCrc = 2892331115;
+    const expectedCrc = 3089006589;
+    asserts.strictEqual(crc, expectedCrc);
+    asserts.notStrictEqual(crc, unexpectedCrc);
+  });
+
+  it(
+    "should record batch encode its attributes and" +
```

Comment:
Please leave a space between "and" and "generate".

Reply:
Solved it